src/org/usfirst/frc/team4571/robot/subsystems/TankDriveSubsystem.java

```diff
@@ -66,38 +68,47 @@ public void initialize() {
 		this.rightEncoder.reset();
 	}
 
-	public double getLeftEncoderDistance() {
-		return Math.abs(this.leftEncoder.getDistance());
-	}
-
-	public double getRightEncoderDistance() {
-		return Math.abs(this.rightEncoder.getDistance());
-	}
-
-	public double getDistance() {
-		double left = getLeftEncoderDistance();
-		double right = getRightEncoderDistance();
-		return Math.max(left, right);
-	}
-
-	public boolean isDriveFinished() {
+//	public double getLeftEncoderDistance() {
+//		return Math.abs(this.leftEncoder.getDistance());
+//	}
+//
+//	public double getRightEncoderDistance() {
+//		return Math.abs(this.rightEncoder.getDistance());
+//	}
+//
+//	public double getDistance() {
+//		double left = getLeftEncoderDistance();
+//		double right = getRightEncoderDistance();
+//		return Math.max(left, right);
+//	}
+
+	public boolean isDriveStraightFinished() {
 		return this.distanceController.onTarget();
 	}
 
 	public double getDisplacement() {
 		return this.encoderAverage.pidGet();
 	}
 
-	public void driveForGivenDistance(double setPoint) {
-		this.distanceController.enable();
-		this.distanceController.setSetpoint(setPoint);
+	public void driveStraightForGivenDistance(double setPoint) {
+		distanceController.reset();
+		distanceController.setInputRange(0, setPoint);
+		distanceController.setOutputRange(-1.0, 1.0);
+		distanceController.enable();
+	}
+	
+	public void drive(double distanceInFeet) {
+		driveStraightForGivenDistance(distanceInFeet);
+		while(isDriveStraightFinished()) {
```

Comment:
This should be !isDriveStraightFinished because you want to drive as long as you havent reached your target